### PR TITLE
[VEN-1709]: simulations for adding a new market in DeFi pool

### PR DESCRIPTION
**Pull request by GitGuru7**
## Description

Resolves VEN-1709

### CODE REVIEW

File: vips/vip-143/vip-143-testnet.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            vankrBNB_DeFi,
+            "0",
+            "0",
+            parseUnits("39", 18),
+            VTOKEN_RECEIVER,
+            parseUnits("5000", 18),
+            parseUnits("4000", 18),
+          ],
+        ],
+      },
```
[the remark this answers is not in the document]
Comment (GitGuru7) [reply vips/vip-143/vip-143-testnet.ts:54]: fixed here [dd604c6](https://github.com/VenusProtocol/vips/pull/41/commits/dd604c60e8972e0f007ac0cc3ba6b238465b81e5)

**Review by chechu — APPROVED** [review]

**Review by chechu — APPROVED** [review]

File: vips/vip-143/vip-143.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import { parseUnits } from "ethers/lib/utils";
+
+import { ProposalType } from "../../src/types";
+import { makeProposal } from "../../src/utils";
+
+const ankrBNB = "0x52F24a5e03aee338Da5fd9Df68D2b6FAe1178827";
+const ANKR = "0xf307910A4c7bbc79691fD374889b36d8531B08e3";
+const POOL_REGISTRY = "0x9F7b01A536aFA00EF10310A162877fd792cD0666";
+const VTOKEN_RECEIVER = "AE1c38847Fb90A13a2a1D7E5552cCD80c62C6508";
```
Comment (chechu) [inline vips/vip-143/vip-143.ts:9]:
```suggestion
const VTOKEN_RECEIVER = "0xAE1c38847Fb90A13a2a1D7E5552cCD80c62C6508";
```

File: simulations/vip-143/vip-143/simulations.ts [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+const USDD = "0xd17479997F34dd9156Deef8F95A52D81D265be9c";
+const vUSDD_DeFi = "0xA615467caE6B9E0bb98BC04B4411d9296fd1dFa0";
+const COMPTROLLER_DeFi = "0x3344417c9360b963ca93A4e8305361AEde340Ab9";
+const BINANCE_ORACLE = "0x594810b741d136f1960141C0d8Fb4a91bE78A820";
+
+const ankrBNB = "0x52F24a5e03aee338Da5fd9Df68D2b6FAe1178827";
+const ANKR = "0xf307910A4c7bbc79691fD374889b36d8531B08e3";
+const POOL_REGISTRY = "0x9F7b01A536aFA00EF10310A162877fd792cD0666";
+const VTOKEN_RECEIVER = "AE1c38847Fb90A13a2a1D7E5552cCD80c62C6508";
```
Comment (chechu) [inline simulations/vip-143/vip-143/simulations.ts:26]:
```suggestion
const VTOKEN_RECEIVER = "0xAE1c38847Fb90A13a2a1D7E5552cCD80c62C6508";
```

File: simulations/vip-143/vip-143/simulations.ts [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+      });
+
+      it("should register vankrBNB_DeFi in PoolRegistry", async () => {
+        const vToken = await poolRegistry.getVTokenForAsset(COMPTROLLER_DeFi, ankrBNB);
+        expect(vToken).to.equal(vankrBNB_DeFi);
+      });
+    });
+
+    describe("Ownership", () => {
+      it("should transfer ownership of vUSDD_Stablecoins to Timelock", async () => {
```
Comment (chechu) [inline simulations/vip-143/vip-143/simulations.ts:122]:
```suggestion
      it("should transfer ownership of vankrBNB_DeFi to Timelock", async () => {
```

File: simulations/vip-143/vip-143/simulations.ts [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+    checkVToken(vankrBNB_DeFi, {
+      name: "Venus ankrBNB (DeFi)",
+      symbol: "vankrBNB_DeFi",
+      decimals: 8,
+      underlying: ankrBNB,
+      exchangeRate: parseUnits("1", 28),
+    });
+  });
+
+  testVip("VIP-143 Add Market", vip143());
```
Comment (chechu) [inline simulations/vip-143/vip-143/simulations.ts:105]: I would check the emitted events

**Review by chechu — APPROVED** [review]

File: simulations/vip-143/vip-143/simulations.ts [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+        });
+
+        describe("RF, CF, LT, Protocol seize share", () => {
+          it("should set vankrBNB_DeFi reserve factor to 0.25", async () => {
+            expect(await vankrBNB.reserveFactorMantissa()).to.equal(parseUnits("0.25", 18));
+          });
+
+          it("should set vankrBNB_DeFi collateral factor to 0", async () => {
+            const market = await comptroller.markets(vankrBNB_DeFi);
+            expect(market.collateralFactorMantissa).to.equal(parseUnits("0", 18));
```
Comment (kkirka) [inline simulations/vip-143/vip-143/simulations.ts:212]:
```suggestion
            expect(market.collateralFactorMantissa).to.equal(0);
```

File: simulations/vip-143/vip-143/simulations.ts [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+          });
+
+          it("should set vankrBNB_DeFi liquidation threshold to 0", async () => {
+            const market = await comptroller.markets(vankrBNB_DeFi);
+            expect(market.liquidationThresholdMantissa).to.equal(parseUnits("0", 18));
```
Comment (kkirka) [inline simulations/vip-143/vip-143/simulations.ts:217]:
```suggestion
            expect(market.liquidationThresholdMantissa).to.equal(0);
```

File: simulations/vip-143/vip-143/simulations.ts [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+            expect(await comptroller.borrowCaps(vankrBNB_DeFi)).to.equal(parseUnits("4000", 18));
+          });
+        });
+
+        describe("Reward Distributor", () => {
+          it("should have 2 rewards distributor in DeFi pool", async () => {
+            const comptroller = await ethers.getContractAt(COMPTROLLER_ABI, COMPTROLLER_DeFi);
+            expect(await comptroller.getRewardDistributors()).to.have.lengthOf(2);
+          });
+
```
Comment (kkirka) [inline simulations/vip-143/vip-143/simulations.ts:241]:
```suggestion

          it("should be added to DeFi pool", async () => {
            expect(await comptroller.getRewardDistributors()).to.include(REWARD_DISTRIBUTOR);
          });

```

File: simulations/vip-143/vip-143/simulations.ts [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+          it("should set vankrBNB_DeFi borrow cap to 200,000", async () => {
+            expect(await comptroller.borrowCaps(vankrBNB_DeFi)).to.equal(parseUnits("4000", 18));
+          });
+        });
+
+        describe("Reward Distributor", () => {
+          it("should have 2 rewards distributor in DeFi pool", async () => {
+            const comptroller = await ethers.getContractAt(COMPTROLLER_ABI, COMPTROLLER_DeFi);
```
Comment (kkirka) [inline simulations/vip-143/vip-143/simulations.ts:238]: We have `comptroller` defined already, so no need to redefine AFAIU
```suggestion
```

**Review by kkirka — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/146

**Pull request merged**